manage.py

```diff
@@ -0,0 +1,179 @@
+
+    default_name = DEFAULT_NAMETAG.format(tag='latest')
+
+    @classmethod
+    def get_full_name(cls, name):
+        return '/'.join((cls.REGISTRY, name))
+
+    def __init__(self, parser):
+        parser.add_argument(
+            '-n', '--name',
+            default=self.default_name,
+            help=f'Image name/tag (default: {self.default_name})',
+        )
+        parser.add_argument(
+            '--label',
+            action='append',
+            help='Additional name/tags to label image; the first of these, '
+                 'if any, is treated as the "version"',
+        )
+        parser.add_argument(
+            '--target',
+            choices=('development', 'production'),
+            default='production',
+            help="Target environment (default: production)",
+        )
+        parser.add_argument(
+            '-l', '--login',
+            action='store_true',
+            help="log in to AWS ECR",
+        )
+        parser.add_argument(
+            '-p', '--push',
+            action='store_true',
+            help="push image once built",
+        )
+        parser.add_argument(
+            '-d', '--deploy',
+            action='store_true',
+            help="deploy the container once the image is pushed",
+        )
+
+    def prepare(self, args, parser):
+        if args.login and not args.push:
+            parser.error("will not log in outside of push operation")
+
+        command = self.local['docker'][
+            'build',
+            '--build-arg', f'TARGET={args.target}',
+            '-t', args.name,
+            '-t', self.get_full_name(args.name),
+        ]
+
+        if args.label:
+            for label in args.label:
+                name = self.DEFAULT_NAMETAG.format(tag=label)
+                command = command[
+                    '-t', name,
+                    '-t', self.get_full_name(name),
+                ]
+
+        yield command[ROOT_PATH]
+
+        if args.push:
+            yield from self['push'].prepare(args)
+
+        if args.deploy:
+            yield self['deploy'].prepare(args)
+
+    @localmethod('-l', '--login', action='store_true', help="log in to AWS ECR")
+    def push(self, args):
+        """push already-built image to registry"""
+        if args.login:
+            login_command = self.local['aws'][
+                'ecr',
+                'get-login',
+                '--no-include-email',
+                '--region', 'us-west-2',
+            ]
+            if args.show_commands or not args.execute_commands:
+                print('>', login_command)
+            if args.execute_commands:
+                full_command = login_command()
+                (executable, *arguments) = full_command.split()
+                assert executable == 'docker'
+                yield self.local[executable][arguments]
+
+        yield self.local['docker'][
+            'push',
+            self.get_full_name(args.name),
+        ]
```

Comment:
...and then pushed it:

    manage build push -l

Though I could've done it all at once:

    manage build --label=0.0.1 -lp